### PR TITLE
New App: Todoist Tasks

### PR DESCRIPTION
**Pull request by TheSchultz**
todoisttasks

Shows up to 3 tasks on your To-Do list, sorted by priority. Use the filter option to further sort your tasks by using parameters such as 'today', tomorrow', 'overdue' etc. 

Let me know if there are any issues. That was all done with ChatGPT4 by modifying existing code. I have no idea how any of this actually works.

### CODE REVIEW

**Comment by TheSchultz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by TheSchultz** [comment]
Cheers @rohansingh The process behind it isn't exactly what I'd call streamlined yet. There were many prompts used and in total all of this was done over 5-6 long conversations back and forth. But it's definitely getting there, it's a great sparring partner. Here's an example of what one of those conversations would look like.  https://shareg.pt/7HT31Iu (Don't mind the user token, was a temporary one I generated for testing)

File: apps/todoisttasks/todoisttasks.star [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+def main(config):
+    token = config.get("auth") or config.get("dev_api_key")
+    children = []
+    circle_children = []  # Initialize circle_children here
+
+    if token:
+        filter = config.get("filter") or DEFAULT_FILTER
+
+        cache_key = "%s/%s" % (token, filter)
+        content = cache.get(cache_key)
```
Comment (matslina) [inline apps/todoisttasks/todoisttasks.star:34]: This cache key is never written to.

**Comment by matslina** [comment]
Curious if this will work, so I'll go ahead and merge. At least have to fix that cache problem before publishing.

**Pull request merged**